hphp/runtime/vm/jit/phys-reg-saver.cpp

```diff
@@ -59,15 +51,13 @@ PhysRegSaver::PhysRegSaver(Vout& v, RegSet regs)
       gpr.forEach([&] (PhysReg r) {
         v << push{r};
       });
+      m_adjust = gpr.size() & 0x1 ? 8 : 0;
       break;
     case Arch::ARM:
       gpr.forEachPair([&] (PhysReg r0, PhysReg r1) {
-        if (r1 == InvalidReg) {
-          v << push{r0};
-        } else {
-          v << pushp{r1, r0};
-        }
+        v << pushp{r1, r0};
```

Comment:
`forEachPair()` nicely returns `InvalidReg` for the second register if it's the final of an odd number of registers. `pushp`/`popp` are designed to recognize InvalidReg and to simply push `$xzr` for ARM and to simply `$sp +/- 8` for the other platforms. Note: I haven't implemented `pushp`/`popp` for X64 and PPC64 yet.